### PR TITLE
[4944] Add a "Help block" in the plugin definition file (craftercms-plugin.yaml) to provide help to users

### PR DESCRIPTION
**Pull request by aachavarria**
adding marked to package.json;
adding documentation prop to MarketplacePlugin interface;

https://github.com/craftercms/craftercms/issues/4944

### CODE REVIEW

File: ui/app/package.json [hunk 1]
```diff
@@ -47,6 +48,7 @@
     "graphql": "^15.5.0",
     "jquery": "^3.5.0",
     "js-cookie": "^2.2.1",
+    "marked": "^3.0.2",
```
Comment (rart) [inline ui/app/package.json:51]: Let's look into using some highlighter for code snippets (e.g. pygmentize or similar). Use same approach for closing lib as with the markdown processor.

File: ui/app/src/components/PluginDetailsView/PluginDetailsView.tsx [hunk 2]
```diff
@@ -315,8 +338,16 @@ export default function PluginDetailsView(props: PluginDetailsViewProps) {
                 />
               </Alert>
             )}
-
             <Typography variant="body1">{description}</Typography>
+            {markdown && <Typography component="div" dangerouslySetInnerHTML={{ __html: markdown }} />}
+            {link && <Link href={link} />}
```
Comment (rart) [inline ui/app/src/components/PluginDetailsView/PluginDetailsView.tsx]: Does the link auto-shows the href as it's visible text?

File: ui/app/src/components/PluginDetailsView/PluginDetailsView.tsx [hunk 3]
```diff
@@ -315,8 +338,16 @@ export default function PluginDetailsView(props: PluginDetailsViewProps) {
...
+            {markdownError && (
+              <ErrorState message={formatMessage(messages.markdownError)}>
+                <Typography>
+                  <Link href={plugin.documentation}>{plugin.documentation}</Link>
+                </Typography>
+              </ErrorState>
```
Comment (rart) [inline ui/app/src/components/PluginDetailsView/PluginDetailsView.tsx]: Let's not make too much fuss about the error. Just stick to a message:

Unable to render documentation. Visit {link} to view.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/PluginDetailsView/PluginDetailsView.tsx (hunk 2 above)
Comment (aachavarria) [reply ui/app/src/components/PluginDetailsView/PluginDetailsView.tsx]: no, it doesn't

**Review by rart — APPROVED** [review]

**Pull request merged**